src/kemal/route_handler.cr

```diff
@@ -49,4 +42,14 @@ class Kemal::RouteHandler < HTTP::Handler
     node = radix_path method, path
     @tree.add node, route
   end
+
+  class RouteNotFoundException < Exception
+    def initialize(method, path)
```

Comment:
I'm confused, above you say the exceptions are not necessary and here you want the context as the param :confused: ?